### PR TITLE
[CLOUDGA-19629] Add support for telemetry provider

### PR DESCRIPTION
**Pull request by aquatiko**
Summary:
This change adds telemetry provider apis.
Almost everything is similar to metric export apis which is based on telemetry CRUD.
I've added IDs in all api outputs and introduced validate command. IDs are used in DB audit apis for specifying telemetry_provider to use.

Test Plan:

Added UTs for the change.

CLI operation logs:
```bash
[rkumar@dev-server-rkumar-31 ybm-cli]$ ./ybm integration create --config-name=dd --type=datadog --datadog-spec=site=datadoghq.com,api-key=f4XXXXXXXXXXXXXXXXXXXXXXXXXXXXbf
A newer version is available. Please upgrade to the latest version v0.1.18
The telemetry provider 51bf0334-4aef-430d-892c-aa840d00dc26 is being created
ID                                     Name      Type      Site            ApiKey
51bf0334-4aef-430d-892c-aa840d00dc26   dd        DATADOG   [datadoghq.com](http://datadoghq.com/)   f4XXXXXXXXXXXXXXXXXXXXXXXXXXXXbf


[rkumar@dev-server-rkumar-31 ybm-cli]$ ./ybm integration update --config-name=dd --new-config-name=ddog --type=datadog --datadogfspec=site=datadoghq.com,api-key=f4XXXXXXXXXXXXXXXXXXXXXXXXXXXXbf
A newer version is available. Please upgrade to the latest version v0.1.18
The telemetryProvider 51bf0334-4aef-430d-892c-aa840d00dc26 is being updated
ID                                     Name      Type      Site            ApiKey
51bf0334-4aef-430d-892c-aa840d00dc26   ddog      DATADOG   [datadoghq.com](http://datadoghq.com/)   f4XXXXXXXXXXXXXXXXXXXXXXXXXXXXbf

[rkumar@dev-server-rkumar-31 ybm-cli]$ ./ybm integration list
A newer version is available. Please upgrade to the latest version v0.1.18
ID                                     Name      Type
51bf0334-4aef-430d-892c-aa840d00dc26   ddog      DATADOG
2c21e4fd-a39c-4958-bfe6-80e727dc2820   ss        DATADOG


[rkumar@dev-server-rkumar-31 ybm-cli]$ ./ybm integration delete --config-name=ddog
A newer version is available. Please upgrade to the latest version v0.1.18
? Are you sure you want to delete config: ddog Yes
Deleting Telemetry Provider ddog


[rkumar@dev-server-rkumar-31 ybm-cli]$ ./ybm integration validate --config-name=dd --type=datadog --datadog-spec=site=datadoghq.com,api-key=f4XXXXXXXXXXXXXXXXXXXXXXXXXXXXbf
A newer version is available. Please upgrade to the latest version v0.1.18
Telemetry provider is valid
```

### CODE REVIEW

File: cmd/telemetry_provider/telemetry_provider.go
Comment (shinji62) [inline cmd/telemetry_provider/telemetry_provider.go:153]: Is deleting emitting a task ?

File: cmd/telemetry_provider/telemetry_provider.go
Comment (shinji62) [inline cmd/telemetry_provider/telemetry_provider.go:209]: is updating emitting a task as welllll ?

File: cmd/telemetry_provider/telemetry_provider.go
Comment (shinji62) [inline cmd/telemetry_provider/telemetry_provider.go:246]: Why do we need a separate validation ? Can we just not vaidate when we create or update?

File: internal/client/client.go
Comment (shinji62) [inline internal/client/client.go:1496]: Just use the same function name.. you be clearer. Same for all others

File: cmd/telemetry_provider/telemetry_provider.go
Comment (aquatiko) [reply cmd/telemetry_provider/telemetry_provider.go:153]: yes, create/update/delete all operations create a new task with ybm-backend

Comment (shinji62) [reply cmd/telemetry_provider/telemetry_provider.go:153]: So in this case usee the wait function. Looks at others like create cluster

File: cmd/telemetry_provider/telemetry_provider.go
Comment (aquatiko) [reply cmd/telemetry_provider/telemetry_provider.go:246]: We have an api to validate telemetry provider config in ybm-backend and all the generated SDKs have that too.
Hence, added an api in CLI as well. 
We do the validations on create/update and return a failed response if the telemetry config is not valid.

CLI use case for this api could be checking a list of configs present with user and select from a valid set based on some prioritisation logic, or maybe some pre-compute step to avoid failing the CRUD operations.

File: cmd/telemetry_provider/telemetry_provider.go
Comment (aquatiko) [reply cmd/telemetry_provider/telemetry_provider.go:153]: @shinji62 sorry for the confusion, the telemetry provider apis dont work under a task. So they dont need a wait function.

The DB audit apis which I added do work under a task and needs to be added with a wait method. I will add that in a separate PR.

File: cmd/db_audit_logs_exporter/db_audit_logs_exporter.go
Comment (jagdish-yb) [inline cmd/db_audit_logs_exporter/db_audit_logs_exporter.go:204]: Not sure but should this text value **I**ntegration with caps I instead of i, WDYT

**Review by jagdish-yb — APPROVED** [review]
Please await code owners approval, thanks.

File: cmd/db_audit_logs_exporter/db_audit_logs_exporter.go
Comment (aquatiko) [reply cmd/db_audit_logs_exporter/db_audit_logs_exporter.go:204]: `[REQUIRED] The ID of the Integration` this part?
the commands are case insensitive afaik `integration-id`

Comment (jagdish-yb) [reply cmd/db_audit_logs_exporter/db_audit_logs_exporter.go:204]: yes - "The ID of the Integration this part"

File: cmd/telemetry_provider/telemetry_provider.go
Comment (shinji62) [reply cmd/telemetry_provider/telemetry_provider.go]: I would love to see that in this PR

File: cmd/telemetry_provider/telemetry_provider.go
Comment (shinji62) [reply cmd/telemetry_provider/telemetry_provider.go]: Not sure I see any value on this valid function ... 
Validation should be done transparently.

File: cmd/telemetry_provider/telemetry_provider.go
Comment (jagdish-yb) [reply cmd/telemetry_provider/telemetry_provider.go]: @shinji62 - This PR is about Telemetry provider lifecycle mgmt. So let's not mix up with the db audit part. 
@aquatiko - For DB Audit part, let's keep the CLI behaviour in sync with what is there for other async operations.

Comment (shinji62) [reply cmd/telemetry_provider/telemetry_provider.go]: @jagdish-yb Ok, so if no task are emitted then that's fine.

**Review by shinji62 — APPROVED** [review]
Still have some reserve on the valid function, but nothing blocking

**Comment by aquatiko** [comment]
> Still have some reserve on the valid function, but nothing blocking

I'll add a commit to remove the validate method and send a total of 2 commits in this PR, so that we can revert this commit if we want and add validate method

File: cmd/telemetry_provider/telemetry_provider.go
Comment (aquatiko) [reply cmd/telemetry_provider/telemetry_provider.go]: removed validation

**Pull request merged**
